scipy/stats/_entropy.py

```diff
@@ -163,40 +205,109 @@ def differential_entropy(
 
     """
     values = np.asarray(values)
-    values = np.moveaxis(values, axis, 0)
+    values = np.moveaxis(values, axis, -1)
+    n = values.shape[-1]  # number of observations
 
     if window_length is None:
-        window_length = math.floor(math.sqrt(len(values)) + 0.5)
+        window_length = math.floor(math.sqrt(n) + 0.5)
 
-    if not 2 <= 2 * window_length < len(values):
+    if not 2 <= 2 * window_length < n:
         raise ValueError(
             f"Window length ({window_length}) must be positive and less "
-            f"than half the sample size ({len(values)}).",
+            f"than half the sample size ({n}).",
         )
 
-    sorted_data = np.sort(values, axis=0)
+    if base is not None and base <= 0:
+        raise ValueError("`base` must be a positive number or `None`.")
 
-    repeats = np.array(
-        (window_length + 1,)
-        + ((1,) * (len(sorted_data) - 2))
-        + (window_length + 1,),
-    )
+    sorted_data = np.sort(values, axis=-1)
 
-    padded_data = np.repeat(
-        sorted_data,
-        repeats=repeats,
-        axis=0,
-    )
+    methods = {"vasicek": _vasicek_entropy,
+               "van es": _van_es_entropy,
+               "correa": _correa_entropy,
+               "ebrahimi": _ebrahimi_entropy,
+               "auto": None}
+    method = method.lower()
+    if method not in methods.keys():
+        message = f"`method` must be one of {set(methods.keys())}"
+        raise ValueError(message)
```

Comment:
I don't think you need to be explicit with the `.key()` method here.

Reply:
You're right. I've written this sort of code several times and never before have I been explicit about `keys`! Not sure what was different in my brain that day : )

Reply:
Pushing fix soon.